pallets/channels/src/set.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright 2019-2021 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
```

Comment:
This file containing the `OrderedSet` is duplicated in `parachain-staking` and `channels`. I considered making ` support` module to put it there so it's not duplicated, but decided aginst it because this PR is already fat (3 new pallets). If you think I should add `support` pallet so this file isn't duplicated, let me know.

Could also be in a follow up. If we put the traits in `support`, we could also remove `token-factory` from `xtransfer` as a dependency and import the `support::traits::TokenMinter` instead. No difference in underlying logic though.

Reply:
And yes, I would prefer to import directly from `orml_utilities`, but that would force us to sync substrate updates with that code thereafter.